tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -1008,3 +1008,16 @@ def test_constant_liar_observation_pairs(direction: str, multivariate: bool) ->
 def test_constant_liar_experimental_warning() -> None:
     with pytest.warns(optuna.exceptions.ExperimentalWarning):
         _ = TPESampler(constant_liar=True)
+
+
+def test_sample_independent_with_branch_division() -> None:
+    sampler = TPESampler(n_startup_trials=0)
+
+    def objective(trial: Trial) -> float:
+        if trial.number < 3:
+            return trial.suggest_float("x", 0, 1)
+        else:
+            return trial.suggest_categorical("c", [0, 1, 2])
```

Comment:
Seems like CI wants you to cast this return value to float https://github.com/optuna/optuna/pull/2739/checks?check_run_id=2789521816.

Reply:
Alternatively, I guess we can replace `suggest_categorical("c", [0, 1, 2])` with `suggest_int("c", 0, 2)`.

Reply:
I`d like to track both the float value and the categorical value here. I will cast as float(trial.suggest_categorical("c", [0, 1, 2])).

Reply:
I fix it as below. PTAL.
```
c = trial.suggest_categorical("c", [0., 1., 2.])
assert isinstance(c, float)
return c
```